include/exadg/solvers_and_preconditioners/preconditioners/preconditioner_amg.h

```diff
@@ -149,10 +149,27 @@ class PreconditionerBoomerAMG : public PreconditionerBase<Number>
   ~PreconditionerBoomerAMG()
   {
 #ifdef DEAL_II_WITH_PETSC
-    PetscErrorCode ierr = VecDestroy(&petsc_vector_dst);
-    AssertThrow(ierr == 0, ExcPETScError(ierr));
-    ierr = VecDestroy(&petsc_vector_src);
-    AssertThrow(ierr == 0, ExcPETScError(ierr));
+    bool const has_subcommunicator =
+      Utilities::MPI::min(system_matrix.m(),
+                          pde_operator.get_matrix_free().get_dof_handler().get_communicator()) == 0;
```

Comment:
just a few minor comments to make this more readable: i) Introduce a variable for the MPI comm to shorten this expression, ii) use brackets for the expression `(... == 0)` right to the assignment, iii) introduce an empty line before the if-statement below